scanner/src/main/kotlin/scanners/scancode/ScanCode.kt

```diff
@@ -114,11 +114,15 @@ class ScanCode internal constructor(
     }
 
     private val scanCodeConfiguration = scannerConfig.options?.get("ScanCode").orEmpty()
+    private val nonSpaceRegex = Regex("\\S+")
 
-    private val configurationOptions = scanCodeConfiguration["commandLine"]?.split(' ')
-        ?: DEFAULT_CONFIGURATION_OPTIONS
-    private val nonConfigurationOptions = scanCodeConfiguration["commandLineNonConfig"]?.split(' ')
-        ?: DEFAULT_NON_CONFIGURATION_OPTIONS
+    private val configurationOptions = scanCodeConfiguration["commandLine"]?.let { commandLine ->
+        nonSpaceRegex.findAll(commandLine).map { it.value }
+    } ?: DEFAULT_CONFIGURATION_OPTIONS
+
+    private val nonConfigurationOptions = scanCodeConfiguration["commandLineNonConfig"]?.let { commandLineNonConfig ->
```

Comment:
Some redundancy could be avoided by moving the Regex processing to a function, maybe an extension function on the map of scanner options, such as
`fun Map<String, String>.commandLineOptions(key: String, defaultOptions: Sequence<String>): Sequence<String>`

Reply:
Your proposal is a tad too specific for my taste, so I came up with a more generic extension function that I can also use in other similar places.